amazon-kinesis-client/src/test/java/software/amazon/kinesis/config/KCLAppConfig.java

```diff
@@ -0,0 +1,197 @@
+                ProfileCredentialsProvider.builder().profileName(getProfile()).build() : DefaultCredentialsProvider.create();
+        builder.credentialsProvider(credentialsProvider);
+
+        return builder.build();
+    }
+
+    default String getWorkerId() throws UnknownHostException {
+        return Inet4Address.getLocalHost().getHostName();
+    }
+
+    default RecordValidatorQueue getRecordValidator() {
+        return new RecordValidatorQueue();
+    }
+
+    default ShardRecordProcessorFactory getShardRecordProcessorFactory() {
+        return new TestRecordProcessorFactory(getRecordValidator());
+    }
+
+    default ConfigsBuilder getConfigsBuilder() throws IOException, URISyntaxException {
+//        return getConfigsBuilder("");
+        final String workerId = getWorkerId();
+        if (getStreamArn() == null) {
+            return new ConfigsBuilder(getStreamName(), getApplicationName(), buildConsumerClient(), buildAsyncDynamoDbClient(),
+                    buildAsyncCloudWatchClient(), workerId, getShardRecordProcessorFactory());
+        } else {
+            return new ConfigsBuilder(Arn.fromString(getStreamArn()), getApplicationName(), buildConsumerClient(), buildAsyncDynamoDbClient(),
+                    buildAsyncCloudWatchClient(), workerId, getShardRecordProcessorFactory());
+        }
+    }
+
+//    default ConfigsBuilder getConfigsBuilder(String workerIdSuffix) throws IOException, URISyntaxException {
+//        final String workerId = getWorkerId() + workerIdSuffix;
+//        if (getStreamArn() == null) {
+//            return new ConfigsBuilder(getStreamName(), getApplicationName(), buildConsumerClient(), buildAsyncDynamoDbClient(),
+//                    buildAsyncCloudWatchClient(), workerId, getShardRecordProcessorFactory());
+//        } else {
+//            return new ConfigsBuilder(Arn.fromString(getStreamArn()), getApplicationName(), buildConsumerClient(), buildAsyncDynamoDbClient(),
+//                    buildAsyncCloudWatchClient(), workerId, getShardRecordProcessorFactory());
+//        }
+//    }
+
+    RetrievalConfig getRetrievalConfig() throws IOException, URISyntaxException;
+
+    /**
+     * Configure ingress load (batch size, record size, and calling interval)
+     */
+    @Value
+    @Builder
+    class ProducerConfig {
+        private boolean isBatchPut;
+        private int batchSize;
+        private int recordSizeKB;
+        private long callPeriodMills;
+    }
+
+    /**
+     * Description of the method of resharding for a test case
+     * <p>
+     * reshardingFactorCycle: lists the scales by which the number of shards in a stream will be updated
+     * in sequence. e.g {2.0, 0.5} means that the number of shards will first be doubled, then halved
+     * <p>
+     * numReshardCycles: the number of resharding cycles that will be executed in a test]
+     * <p>
+     * reshardFrequencyMillis: the period of time between reshard cycles (in milliseconds)
```

Comment:
FMI: what happens if the stream is still being resharded (by Kinesis) when the next "reshard cycle", controlled by this config, happens? Will this code fail, or retry?

Reply:
This will cause this individual reshard to fail (not the whole cycle) and it will move on to the next reshard in the cycle (or next cycle, if at the end of the current cycle).